MovieApp/MovieApp/App/NetworkDataSource/MovieNetworkDataSource.swift

```diff
@@ -20,14 +20,68 @@ class MovieNetworkDataSource: MovieNetworkDataSourceProtocol {
         }
     }
     
+    func fetchTrendingToday(completion: @escaping(Result<[MovieDataModel]?, RequestError>) -> Void) {
+        movieClient.fetchTrendingToday { [weak self] result in
+            switch result {
+            case .success(let movies):
+                let mappedMovies = self?.mapMovies(movies)
+                completion(.success(mappedMovies))
+            case .failure(let error):
+                completion(.failure(error))
+            }
+        }
+    }
+    
+    func fetchTrendingThisWeek(completion: @escaping(Result<[MovieDataModel]?, RequestError>) -> Void) {
+        movieClient.fetchTrendingThisWeek { [weak self] result in
+            switch result {
+            case .success(let movies):
+                let mappedMovies = self?.mapMovies(movies)
+                completion(.success(mappedMovies))
+            case .failure(let error):
+                completion(.failure(error))
+            }
+        }
+    }
+    
+    func fetchTopRatedMovies(completion: @escaping(Result<[MovieDataModel]?, RequestError>) -> Void) {
+        movieClient.fetchTopRatedMovies { [weak self] result in
+            switch result {
+            case .success(let movies):
+                let mappedMovies = self?.mapMovies(movies)
+                completion(.success(mappedMovies))
+            case .failure(let error):
+                completion(.failure(error))
+            }
+        }
+    }
```

Comment:
You can create one mapper function for this.
Also, you might not need `mapMovies` function since you'll have one function that will handle the switch case.